tests/unit_tests/test_json_transitions.py

```diff
@@ -0,0 +1,252 @@
+                        "rule_id": "xccdf_org.ssgproject.content_rule_rpm_verify_hashes",
+                        "result": "pass",
+                    },
+                    "rule_a": {
+                        "rule_id": "rule_a",
+                        "result": "notselected",
+                    },
+                }
+            },
+            [],
+            {
+                "rules": {
+                    "xccdf_org.ssgproject.content_rule_rpm_verify_hashes": {
+                        "rule_id": "xccdf_org.ssgproject.content_rule_rpm_verify_hashes",
+                        "result": "pass",
+                    },
+                }
+            },
+        ),
+        (
+            {
+                "rules": {
+                    "xccdf_org.ssgproject.content_rule_rpm_verify_hashes": {
+                        "rule_id": "xccdf_org.ssgproject.content_rule_rpm_verify_hashes",
+                        "result": "pass",
+                    },
+                    "rule_a": {
+                        "rule_id": "rule_a",
+                        "result": "fail",
+                    },
+                }
+            },
+            ["xccdf_org.ssgproject.content_rule_rpm_verify_hashes"],
+            {
+                "rules": {
+                    "xccdf_org.ssgproject.content_rule_rpm_verify_hashes": {
+                        "rule_id": "xccdf_org.ssgproject.content_rule_rpm_verify_hashes",
+                        "result": "pass",
+                    },
+                }
+            },
+        ),
+        (
+            {
+                "rules": {
+                    "xccdf_org.ssgproject.content_rule_rpm_verify_hashes": {
+                        "rule_id": "xccdf_org.ssgproject.content_rule_rpm_verify_hashes",
+                        "result": "pass",
+                    },
+                    "rule_a": {
+                        "rule_id": "rule_a",
+                        "result": "fail",
+                    },
+                }
+            },
+            ["xccdf_org.ssgproject.content_rule_rpm_verify_hashes", "rule_a"],
+            {
+                "rules": {
+                    "xccdf_org.ssgproject.content_rule_rpm_verify_hashes": {
+                        "rule_id": "xccdf_org.ssgproject.content_rule_rpm_verify_hashes",
+                        "result": "pass",
+                    },
+                    "rule_a": {
+                        "rule_id": "rule_a",
+                        "result": "fail",
+                    },
+                }
+            },
+        ),
+    ],
+)
+def test_remove_not_selected_rules(dictionary_json, ids_of_selected_rules, result):
+    remove_not_selected_rules(dictionary_json, ids_of_selected_rules)
+    assert dictionary_json == result
+
+
+@pytest.mark.unit_test
+@pytest.mark.parametrize(
+    "dictionary, result",
+    [
+        ({"a": "", "b": {"c": ""}, "x": "x"}, {"x": "x"}),
```

Comment:
You're testing the same `"b": {"c": ""}` 5 times. It does not bring any more security but drastically reduces test readability. Make sure that your cases are as unique as possible.